File: leetcode/easy/ArrayRotateImage.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        # using zip
        matrix[:] = map(list, zip(*matrix[::-1]))
        # using for
        matrix[:] = matrix[::-1]
            for i in range(len(matrix)):
                for j in range(i):
                    matrix[i][j], matrix[j][i] = matrix[j][i], matrix[i][j]
    

File: leetcode/easy/ArrayRotateImage.js
var rotate = function(matrix) {
    matrix.reverse()
    let temp;
    for (let i=0; i<matrix.length ; i++) {
        for (let j=0; j<i ; j++) {
            temp = matrix[i][j];
            matrix[i][j] = matrix[j][i];
            matrix[j][i] = temp;
        }
    }
    return matrix;
};
